iset.mm

```diff
@@ -151276,82 +151357,85 @@ the additive structure must be abelian (see ~ ringcom ), care must be
-    lringring $p |- ( R e. LRing -> R e. Ring ) $=
-      ( vx vy vr clring wcel crg cur cfv c0g wne cv cplusg co wceq cui cbs wral
-      wo fveq2 wi neeq12d oveqd eqeq12d eleq2d orbi12d imbi12d anbi12d df-lring
-      wa raleqbidv elrab2 simplbi ) AEFAGFAHIZAJIZKZBLZCLZAMIZNZUNOZUQAPIZFZURV
-      BFZSZUAZCAQIZRZBVGRZUJZDLZHIZVKJIZKZUQURVKMIZNZVLOZUQVKPIZFZURVRFZSZUAZCV
-      KQIZRZBWCRZUJVJDAGEVKAOZVNUPWEVIWFVLUNVMUOVKAHTZVKAJTUBWFWDVHBWCVGVKAQTZW
-      FWBVFCWCVGWHWFVQVAWAVEWFVPUTVLUNWFVOUSUQURVKAMTUCWGUDWFVSVCVTVDWFVRVBUQVK
-      APTZUEWFVRVBURWIUEUFUGUKUKUHBCDUIULUM $.
+    $d R r x y $.  $d B r x y $.  $d .+ r $.  $d .1. r $.  $d U r $.
+    islring.b $e |- B = ( Base ` R ) $.
+    islring.a $e |- .+ = ( +g ` R ) $.
+    islring.1 $e |- .1. = ( 1r ` R ) $.
+    islring.u $e |- U = ( Unit ` R ) $.
+    $( The predicate "is a local ring".  (Contributed by SN, 23-Feb-2025.) $)
+    islring $p |- ( R e. LRing <-> ( R e. NzRing /\
+      A. x e. B A. y e. B ( ( x .+ y ) = .1. -> ( x e. U \/ y e. U ) ) ) ) $=
+      ( vr cv cplusg cfv cur wcel wral fveq2 eqtr4di co wceq cui wo wi cbs cnzr
+      clring oveqd eqeq12d eleq2d orbi12d imbi12d raleqbidv df-lring elrab2 ) A
+      MZBMZLMZNOZUAZUSPOZUBZUQUSUCOZQZURVDQZUDZUEZBUSUFOZRZAVIRUQURDUAZGUBZUQFQ
+      ZURFQZUDZUEZBCRZACRLEUGUHUSEUBZVJVQAVICVRVIEUFOCUSEUFSHTZVRVHVPBVICVSVRVC
+      VLVGVOVRVAVKVBGVRUTDUQURVRUTENODUSENSITUIVRVBEPOGUSEPSJTUJVRVEVMVFVNVRVDF
+      UQVRVDEUCOFUSEUCSKTZUKVRVDFURVTUKULUMUNUNABLUOUP $.
   $}
 
   ${
     $d R r x y $.
+    $( A local ring is a nonzero ring.  (Contributed by SN, 23-Feb-2025.) $)
+    lringnzr $p |- ( R e. LRing -> R e. NzRing ) $=
+      ( vx vy vr clring cnzr cv cplusg cfv co cur wceq cui wcel wo cbs df-lring
+      wi wral ssrab3 sseli ) EFABGZCGZDGZHIJUDKILUBUDMIZNUCUENORCUDPIZSBUFSDFEB
+      CDQTUA $.
+  $}
+
+  $( A local ring is a ring.  (Contributed by Jim Kingdon, 20-Feb-2025.)
+     (Revised by SN, 23-Feb-2025.) $)
+  lringring $p |- ( R e. LRing -> R e. Ring ) $=
+    ( clring wcel cnzr crg lringnzr nzrring syl ) ABCADCAECAFAGH $.
+
+  ${
+    lringnz.1 $e |- .1. = ( 1r ` R ) $.
+    lringnz.2 $e |- .0. = ( 0g ` R ) $.
     $( A local ring is a nonzero ring.  (Contributed by Jim Kingdon,
-       20-Feb-2025.) $)
-    lringnz $p |- ( R e. LRing -> ( 1r ` R ) =/= ( 0g ` R ) ) $=
-      ( vx vy vr clring wcel cur cfv c0g wne cv cplusg co wceq cui wo cbs fveq2
-      wi wral wa neeq12d oveqd eqeq12d eleq2d orbi12d imbi12d raleqbidv anbi12d
-      crg df-lring elrab2 simprbi simpld ) AEFZAGHZAIHZJZBKZCKZALHZMZUPNZUSAOHZ
-      FZUTVDFZPZSZCAQHZTZBVITZUOAUJFURVKUAZDKZGHZVMIHZJZUSUTVMLHZMZVNNZUSVMOHZF
-      ZUTVTFZPZSZCVMQHZTZBWETZUAVLDAUJEVMANZVPURWGVKWHVNUPVOUQVMAGRZVMAIRUBWHWF
-      VJBWEVIVMAQRZWHWDVHCWEVIWJWHVSVCWCVGWHVRVBVNUPWHVQVAUSUTVMALRUCWIUDWHWAVE
-      WBVFWHVTVDUSVMAORZUEWHVTVDUTWKUEUFUGUHUHUIBCDUKULUMUN $.
+       20-Feb-2025.)  (Revised by SN, 23-Feb-2025.) $)
+    lringnz $p |- ( R e. LRing -> .1. =/= .0. ) $=
```

Comment:
You ask whether there's anything iset.mm-specific about whether to have `.1.` and `.0.` variables. The answer is basically that there isn't. iset.mm would have to substitute in `eqid` more often but set.mm would but the tradeoffs aren't vastly different between the two and because there is a `R e. LRing` antecedent here, the `eqid` trick will be possible.